api/v1alpha1/queuedworkload_types.go

```diff
@@ -39,6 +39,13 @@ type QueuedWorkloadSpec struct {
 
 	// admission holds the parameters of the admission of the workload by a ClusterQueue.
 	Admission *Admission `json:"admission,omitempty"`
+
+	// priority determines the order of access to the resources in the
+	// ClusterQueue associated with the queue where the workload is located.
+	// The higher the value, the higher the priority.
+	// It's ClusterQueue-scoped and generated according to the priorityclass of the
```

Comment:
> generated according to the priorityclass

hmm, this is not really enforced by Kueue, so it can't be claimed. Remember that the job controller we have is not really part of Kueue.

I think this field should be a PriorityClassName, not the priority int itself; the queuedworkload-controller reads the actual priority from the class and make it available to Kueue internals.

Reply:
SG. I can store the actual priority into the internals info of workload.  However, during the workload queuing process, the user can delete and recreate the priority classes with different values. Perhaps the value of the workload can be refreshed each time re-queue.

Reply:
I think we should indeed have the priority value here too, as we do in PodSpec. Ideally it would be loaded at creation, but since we don't have a webhook yet, we might need to do it in the queuedworkload-controller for now.

Reply:
Having the actual priority on the workload spec means we need a way to limit setting it directly to prevent circumventing priority class (this is done for pods in by the priority admission controller); so we need a webhook to enforce that.

One more thing, we need to limit which namespaces can use which PriorityClasses, this is done via quota scopes in k8s at the pod level; to enforce it at the QueuedWorkload level, we need a webhook as well.

Reply:
I find it's difficult to add the priority as the field of workload spec if we don't have an admission web-hook to populate priority from priority class. When queued-workload-controller receive the create event of workload, the workload is already created, we don't have chance to populate priority from priority class unless we update the workload again on `create func`. 

In comparison, it would make more sense to pass Priority as a parameter in the initialization of workload info as an internal data. WDYT? @ahg-g @alculquicondor

Reply:
I would continue with the simpler approach of looking up the priority at the time we add the workload to the queue. 

The idea is to show how this whole thing works, but we keep the issue open to tighten the API.